### DISCUSSION TITLE
Vector Search

### DISCUSSION THREAD

**Discussion by somitBond** [Automattic/mongoose#13994] [Q&A]
When will we get Vector Search implementation?

**Comment by vkarpov15** [comment]
Vector search in MongoDB is just an aggregation stage, so you can use it with Mongoose's `Model.aggregate()` function

**Reply by Jeremy-Altimea** [reply]
But knnBeta is deprecated.
Do yo have an example ? @vkarpov15

**Comment by IslandRhythms** [comment]
Here is a script for Vector Search Implementation
```javascript
const mongoose = require('mongoose');

const vectorSchema = new mongoose.Schema({
  plot: String,
  genres: [String],
  year: Number,
  myVector: [Number]
});

const Embedded = mongoose.model('vectortest', vectorSchema, 'vectortest');

async function run() {
  await mongoose.connect('');
  // await mongoose.connection.dropDatabase();

  await Embedded.create({
    plot: 'Test plot A',
    genres: ['Action', 'Adventure', 'Family'],
    year: 1993,
    myVector: [0.75, 0.1]
  });

  const agg = [
    {
      '$vectorSearch': {
        'index': 'vector_index',
        'path': 'myVector',
        'queryVector': [0.75, 0.1], // Adjust the query vector as needed
        'numCandidates': 6,
        'limit': 1
      }
    },
    {
      '$project': {
        '_id': 0,
        'plot': 1,
        'genres': 1,
        'year': 1,
        'score': {
          '$meta': 'vectorSearchScore'
        }
      }
    }
  ];

  const res = await Embedded.aggregate(agg);
  console.log('Search result:', res);
  console.log('Done');
}

run();
```

**Reply by WillCalderwood** [reply]
This will only work on an instance hosted on Atlas I assume?

**Reply by vkarpov15** [reply]
@WillCalderwood yes, AFAIK vector search is Atlas only, no vector search on self-hosted MongoDB

**Reply by Jervx** [reply]
I confirm that this works Mongodb Atlas. Is there any way to set this index in mongoose model? Or this is only atlas specific?

![image](https://github.com/user-attachments/assets/d43021c8-d4f7-4201-a514-a05619b36668)

**Reply by vkarpov15** [reply]
@Jervx I don't think you can set this index in mongoose model. Vector indexes are Atlas specific.

**Reply by uratmangun** [reply]
> @Jervx I don't think you can set this index in mongoose model. Vector indexes are Atlas specific.

😭😭😭😭 thanks for the information tho @vkarpov15
